File: leetcode/Merge Sorted Array.cpp
#define F(i, a, b) for(int i = a; i <= b; i++)
#include <vector>
#include <unordered_set>

class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        if( n == 0 ){
            return ;
        }

        int index = m+n-1;
        m--;
        n--;
        while( index >= 0 ){
            if( m < 0 ){
                A[index] = B[n--]; 
            }
            else if( n < 0 ){
                A[index] = A[m--];
            }
            else if( A[m] > B[n] ){
                A[index] = A[m--];
            }
            else{
                A[index] = B[n--];            
            }
            index--;
        }

        return ;
    }
};

File: LeetCode-Java/Merge Sorted Array.java
public class Solution {
	public static void merge(int A[], int m, int B[], int n) {
		int i = m + n-1;
		m--;
		n--;
		while( i >= 0 ){
			if( m < 0 ){
				while( n >= 0 ){
					A[i--] = B[n--];
				}
			}else{
				if( n < 0 ){
					while( m >= 0 ){
						A[i--] = A[m--];
					}
				}else{
					if( A[m] > B[n] ){
						A[i--] = A[m--];
					}else{
						A[i--] = B[n--];
					}
				}
			}
		}

		return ;
	}
}
